tests/test_accumulator_fixed.rs

```diff
@@ -11,24 +11,24 @@ use simple_accumulator::SimpleAccumulator;
 fn test_sanity_push_in_fixed_capacity() {
     const CAPACITY: usize = 3;
     println!("Creating...");
-    let mut acc = SimpleAccumulator::with_fixed_capacity::<f64>(&[], CAPACITY, true);
+    let mut acc = SimpleAccumulator::with_fixed_capacity(&[], CAPACITY);
     println!("{acc:?}");
 
     let data = vec![0.0, 1.1, 2.2, 3.3, 4.4];
     for &v in &data {
         acc.push(v);
     }
     println!("{acc:?}");
-    assert_eq!(acc.vec.len(), CAPACITY);
-    assert_eq!(acc.vec, vec![3.3, 4.4, 2.2]);
+    assert_eq!(acc.len(), CAPACITY);
+    assert_eq!(acc.data, vec![2.2, 3.3, 4.4]);
```

Comment:
Adding this after line 23 fails: `assert_eq!(acc.mean(), 3.3);`

Reply:
You are comparing two floats. Use specific crate designed to compare floats e.g. `float_eq`.

Reply:
@dilawar 

```
thread 'test_sanity_push_in_fixed_capacity' panicked at tests/test_accumulator_fixed.rs:24:5:
assertion `left == right` failed
  left: 2.2
 right: 3.3
```

Reply:
Adds a test shows that the running mean is equal to the "dense" mean.

Reply:
> **https://github.com/SubconsciousCompute/SimpleAccumulator/pull/22#discussion_r1433583116**

I've replaced Vec with VecDeque. `push_back` and `pop_front` are very cheap. This simplifies are API. We can bring back the `Vec` based implementation but it has to be encapsulated as `RingBuffer` or `CircularBuffer`.

Reply:
> > **[#22 (comment)](https://github.com/SubconsciousCompute/SimpleAccumulator/pull/22#discussion_r1433583116)**
> 
> I've replaced Vec with VecDeque. `push_back` and `pop_front` are very cheap. This simplifies are API. We can bring back the `Vec` based implementation but it has to be encapsulated as `RingBuffer` or `CircularBuffer`.

We were using Vec as ring buffer, we had a current read and current write position same as any ring buffer, ours was also less expensive then before, pop_front is more expensive than simple overwrite of existing value https://doc.rust-lang.org/src/alloc/collections/vec_deque/mod.rs.html#1583

Reply:
We can do a benchmark and bring back the old implementation or any other faster implementation. Let's figure out the public API.